src/Microsoft.Health.Checkpoints/ICheckpointClient.cs

```diff
@@ -0,0 +1,16 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Checkpoints
+{
+    public interface ICheckpointClient
+    {
+        Task<ICheckpoint> GetCheckpointAsync(string partition, string checkpointIdentifier);
```

Comment:
nit: cancellation token support on async methods (and interfaces) something like `CancellationToken token = default`.

Reply:
updated